File: 12.c
/*Project Euler solution 12, author: http://github.com/TerkaSlan */
#include <stdio.h>
#include <time.h>

static unsigned check_divisors(unsigned long num){
    unsigned long i;
    unsigned ret = 1;
    for(i=2; i<=num; i++){
        unsigned count = 0;
        while(num%i==0){
            num /= i;
            count++;
        }
        ret*=count+1;
    }
    return ret;
}

int main()
{
    clock_t start, end; float time_passed;
    start = clock();
    unsigned long num=1, i=1;
    while(check_divisors(num)<500){
        num+=++i;
    }
    end = clock();
    time_passed = (end-start)*1.0/CLOCKS_PER_SEC;
    printf("%lu\n", num);
    printf("It took me %f seconds\n", time_passed); //0.327864s
    return 0;
}

File: 12.py
#Project Euler solution 12, author: http://github.com/TerkaSlan #
from time import clock

def num_divisors(num):
    if num%2==0:  num /= 2
    count=0
    divisors = 1
    while num%2==0:
        num /= 2
        count+=1
    divisors *= (count+1)
    p=3
    while num!=1:
        count = 0
        while num%p == 0:
            count+=1
            num /= p
        divisors *= (count+1)
        p+=2
    return divisors

def return_index(limit):
    n = 1
    num_1, num_2 = num_divisors(n), num_divisors(n+1)
    while num_1*num_2 < limit:
        n += 1
        num_1, num_2 = num_2, num_divisors(n+1)
    return n

start = clock()
index = return_index(500)
triangle = (index * (index + 1)) / 2
end = clock()
print 'Result: ', triangle
print 'SOlution took me ', end-start, ' seconds'
